roles/create_nginx_app_with_backup/tasks/main.yml

```diff
@@ -3,34 +3,33 @@
   k8s:
     state: "{{ state }}"
     definition: "{{ lookup('file', 'base.yaml')}}"
+
 - name: Wait 2 minutes for nginx deployment
-  debug:
-    var: deploy
-  until: deploy and deploy.status.replicas == deploy.status.get("availableReplicas", 0)
-  retries: 10
-  delay: 10
-  vars:
-    deploy: '{{ lookup("k8s",
-      kind="Deployment",
-      api_version="apps/v1",
-      namespace="nginx-example",
-      label_selector="app=nginx"
-    )}}'
+  k8s_facts:
+    api_version: "apps/v1"
+    kind: "Deployment"
+    namespace: "nginx-example"
+    label_selectors:
+      - "app = nginx"
+  register: deploy
+  until: deploy
+         and (deploy.resources | length > 0)
+         and (deploy.resources[0].get("status", {}).get("replicas",-1) == deploy.resources[0].get("status", {}).get("availableReplicas", -2))
```

Comment:
Just curious, what is the second argument to `get()`? Why use `-1` and `-2`?

Reply:
Just kidding saw your comments in slack. It's the python `get` which allows us to set a default. Looks great!